smart/map_processing.py

```diff
@@ -0,0 +1,171 @@
+import numpy as np
+import skimage as ski
+from matplotlib import colors
+from skimage.morphology import disk
+
+import astropy.units as u
+
+from sunpy.map import Map, all_coordinates_from_map, coordinate_is_on_solar_disk
+
+
+def map_processing(file):
+    """
+    Creating a map from a fits file and processing said map.
+
+    Parameters
+    ----------
+    file : fits
+        fits file to turn into map.
+
+    Returns
+    -------
+    im_map : sunpy.map.Map
+        processed map created from fits file.
+
+    """
+    im_map = Map(file)
+    if im_map.meta["CROTA2"] != 0:
+        im_map = im_map.rotate(order=3)
+    im_map.data[~coordinate_is_on_solar_disk(all_coordinates_from_map(im_map))] = np.nan
+    im_map.cmap.set_bad("k")
+    im_map.plot_settings["norm"] = colors.Normalize(vmin=-200, vmax=200)
+    return im_map
+
+
+@u.quantity_input
+def STL(
+    im_map,
+    thresh=100,
+    dilation_radius: u.arcsec = 9000 * u.arcsec,
+    sigma: u.arcsec = 29000 * u.arcsec,
+    min_size: u.arcsec = 9000000 * u.arcsec,
+):
+    """
+
+    We apply Smoothing, a noise Threshold, and an LOS correction, respectively, to the data.
+
+    Parameters
+    ----------
+    im_map : sunpy.map.Map
+        Processed SunPy magnetogram map.
+    thresh : int, optional
+        Threshold value (in Gauss) to identify regions of interest (default is 100).
+    dilation_radius : int, optional
+        Radius of the disk for binary dilation (default is 9000 arcsecs).
+    sigma : int, optional
+        Standard deviation for Gaussian smoothinß (default is 29000 arcsecs).
+    min_size : int, optional
+        Minimum size of regions to keep in final mask (default is 9000000 arcsecs**2).
+
+    Returns
+    -------
+    smooth_map : sunpy.map.Map
+        Map after applying Gaussian smoothing.
+    filtered_labels : numpy.ndarray
+        2D array with each pixel labelled.
+    mask_sizes : numpy.ndarray
+        Boolean array indicating the sizes of each labeled region.
+
+    """
+
+    arcsec_to_pixel = 1800 * (im_map.scale[0] + im_map.scale[1])
+    dilation_radius = (np.round(dilation_radius / arcsec_to_pixel)).value
+    sigma = (np.round(sigma / arcsec_to_pixel)).value
+    min_size = (np.round(min_size / arcsec_to_pixel)).value
+
+    cosmap_data = cosine_correction(im_map)[0]
```

Comment:
Are the other things returned from `cosine_correction` ever used? If not might make sense to only return the first one?